File: src/main/java/compiler/tokenizer/TokenType.java
package compiler.tokenizer;

public enum TokenType {

	/** Absolutely nothing. */
	EMPTY,
	
	/** A token. For example, ( ) = , */
	TOKEN,
	
	/** First character is a letter, any proceeding characters are letters or numbers. */
	IDENTIFIER,
	
	/** Integer. */
	INTEGER_LITERAL,

	/** Double */
	DOUBLE_LITERAL,
	
	/** Anything enclosed in double quotes. "Hello" "1" */
	STRING_LITERAL,

	/** Add operator **/
	ADD_OPERATOR,

	/** Subtract operator **/
	SUBTRACT_OPERATOR,

	/** Multiply operator **/
	MULTIPLY_OPERATOR,

	/** Divide operator **/
	DIVIDE_OPERATOR,

	/** End a statement **/
	END_STATEMENT,

	/** Colon **/
	COLON,

	/** IfBlock statement **/
	IF_STATEMENT,

	/** Smaller than **/
	SMALLER_THAN,

	/** Smaller than or equal to **/
	SMALLER_THAN_EQUAL,

	/** Larger than **/
	LARGER_THAN,

	/** Larger than or equal to **/
	LARGER_THAN_EQUAL,

	/** Equal to **/
	EQUAL_TO,

	/* Statements for  while loop */
	WHILE_STATEMENT


}

File: src/main/scala/compiler/tokenizer/TokenType.scala
/*
 * Cobalt Programming Language Compiler
 * Copyright (C) 2017  Cobalt
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package compiler.tokenizer

object TokenType extends Enumeration {
  type TokenType = Value
  val EMPTY,
  TOKEN,
  IDENTIFIER,
  RETURN_TYPE,

  INTEGER_LITERAL,
  DOUBLE_LITERAL,
  FLOAT_LITERAL,
  STRING_LITERAL,
  CHARACTER_LITERAL,

  MODULUS_OPERATOR,
  ADD_OPERATOR,
  SUBTRACT_OPERATOR,
  MULTIPLY_OPERATOR,
  DIVIDE_OPERATOR,
  END_STATEMENT,
  COLON,
  SMALLER_THAN,
  SMALLER_THAN_EQUAL,
  LARGER_THAN,
  LARGER_THAN_EQUAL,
  EQUAL_TO = Value
}